### PR TITLE
Add pprof protobuf export into perf profile source connector. 

### PR DESCRIPTION
**Pull request by etep**
Summary: We add a method `CreatePProfProfile` to convert from the perf profiler output to a pprof protobuf.

Type of change: /kind feature

Test Plan: Tested locally with the `stirling_profiler` binary. The profiler binary, after finishing, writes a pprof .pb file to disk. The file, when opened with `go tool pprof` renders the expected flamegraph.

### CODE REVIEW

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        // Add the string to the string table in the profile.
+        profile.add_string_table(std::string(symbol));
+
+        // Add the new location-id into the sample in the profile.
+        sample->add_location_id(location_id);
+
+        // Record the fact that we have written this string into the profile. Save the
+        // location-id because it is used if we find this symbol again.
+        strings[symbol] = location_id;
```
Comment (ddelnano) [inline src/stirling/source_connectors/perf_profiler/pprof/pprof.cc:135]: Isn't this covered by the `try_emplace` [here](https://github.com/pixie-io/pixie/pull/1269/files#diff-64e85fe1b081085e3ae5f47fbf8e1f768cad29707d3bb7d4267ca52af7f170c4R93)?

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  profile.add_string_table("cpu");
+  profile.add_string_table("nanoseconds");
+
+  // State variables useful as we build the pprof profile message.
+  // No locations messages exist (yet); next_location_id starts at 1.
+  // We already added some strings; next_string_id starts at 5.
+  uint64_t next_location_id = 1;
+  uint64_t next_string_id = 5;
+
+  // To build the profile, we iterate over the stack traces  histogram.
```
Comment (ddelnano) [inline src/stirling/source_connectors/perf_profiler/pprof/pprof.cc:68]: >   // To build the profile, we iterate over the stack traces histogram.

There is an extra space between "traces" and "histogram".

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    const std::vector<std::string_view> symbols = absl::StrSplit(stack_trace_str, ";");
+
+    // Iterate over the symbols in this stack trace.
+    // Each symbol will be added to the sample as a "location" message, which in turn refers to a
+    // string in the strings table (tracked in map "strings") (more details below).
+    // Note, because of how we built the stack trace string and how the pprof profile is
+    // organized, we iterate in reverse order.
+    for (auto symbols_iter = symbols.rbegin(); symbols_iter != symbols.rend(); ++symbols_iter) {
+      // For convenience.
+      const auto& symbol = *symbols_iter;
```
Comment (ddelnano) [inline src/stirling/source_connectors/perf_profiler/pprof/pprof.cc:89]: Feel free to drop. Given the depth of the other comments and usefulness, I think this one could be removed.

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+      // try_emplace() looks up an existing entry in the strings table, or creates a new entry
+      // with the key provided. Here a new entry maps from symbol to next_location_id.
+      const auto [strings_iter, inserted] = strings.try_emplace(symbol, next_location_id);
+
+      if (inserted) {
+        // New symbol: create a new location, add it to the sample, and create a new symbol.
+
+        // For convenience & clarity, store the ids in locals, then compute their next values.
```
Comment (ddelnano) [inline src/stirling/source_connectors/perf_profiler/pprof/pprof.cc:98]: Same thing here. I think this comment could be omitted, but feel free to drop my suggestion.

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc (hunk 1 above)
Comment (etep) [reply src/stirling/source_connectors/perf_profiler/pprof/pprof.cc]: Yes, good point. Fixed.

File: src/stirling/source_connectors/perf_profiler/pprof/pprof.cc (hunk 2 above)
Comment (etep) [reply src/stirling/source_connectors/perf_profiler/pprof/pprof.cc]: Yikes, thanks for noticing!

**Review by ddelnano — APPROVED** [review]

File: src/stirling/source_connectors/perf_profiler/pprof/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+load("@rules_cc//cc:defs.bzl", "cc_proto_library")
+load("//bazel:pl_build_system.bzl", "pl_cc_library")
+
+package(default_visibility = ["//src/stirling:__subpackages__"])
+
+cc_proto_library(
```
[the remark this answers is not in the document]
Comment (JamesMBartlett) [reply src/stirling/source_connectors/perf_profiler/pprof/BUILD.bazel:22]: `pl_cc_proto_libarry` apparently doesn't work with external protos

**Review by JamesMBartlett — APPROVED** [review]

**Pull request merged**